projects/plugins/jetpack-mu-wpcom/src/features/coming-soon/coming-soon.php

```diff
@@ -0,0 +1,235 @@
+ *
+ * @return bool
+ */
+function is_accessed_by_valid_share_link( $share_code ) {
+	$preview_links_option = get_option( 'wpcom_public_preview_links' );
+
+	if ( ! is_array( $preview_links_option ) || ! count( $preview_links_option ) || ! $share_code ) {
+		return false;
+	}
+
+	if ( ! in_array( $share_code, array_column( $preview_links_option, 'code' ), true ) ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Add X-Robots-Tag header to prevent from indexing page that includes share code.
+ *
+ * @param array $headers Headers.
+ * @return array Headers.
+ */
+function maybe_add_x_robots_headers( $headers ) {
+	if ( get_share_code() ) {
+		$headers['X-Robots-Tag'] = 'noindex, nofollow';
+	}
+	return $headers;
+}
+add_filter( 'wp_headers', __NAMESPACE__ . '\maybe_add_x_robots_headers' );
+
+/**
+ * Track site preview event.
+ *
+ * @return void
+ */
+function track_preview_link_event() {
+	$event_name = 'wpcom_site_previewed';
+	if ( function_exists( 'wpcomsh_record_tracks_event' ) ) {
+		wpcomsh_record_tracks_event( $event_name, array() );
+	} else {
+		require_lib( 'tracks/client' );
+		tracks_record_event( get_current_user_id(), $event_name, array() );
+	}
+}
+
+/**
+ * Returns the WP.com logo
+ *
+ * @return string
+ */
+function coming_soon_share_image() {
+	return 'https://s1.wp.com/home.logged-out/images/wpcom-og-image.jpg';
+}
+
+/**
+ * Renders a fallback coming soon page
+ */
+function render_fallback_coming_soon_page() {
+	if ( ! defined( 'GRAVATAR_HOVERCARDS__DISABLE' ) ) {
+		define( 'GRAVATAR_HOVERCARDS__DISABLE', true );
+	}
+
+	// Disable WP scripts, likes, social og meta, cookie banner.
+	remove_action( 'wp_enqueue_scripts', 'wpcom_actionbar_enqueue_scripts', 101 );
+	remove_action( 'wp_head', 'print_emoji_detection_script', 7 );
+	remove_action( 'wp_print_styles', 'print_emoji_styles' );
+	remove_action( 'wp_head', 'header_js', 5 );
+	remove_action( 'wp_head', 'global_css', 5 );
+	remove_action( 'wp_footer', 'wpcom_subs_js' );
+	remove_action( 'wp_footer', 'stats_footer', 101 );
+	add_filter( 'infinite_scroll_archive_supported', '__return_false', 99 ); // Disable infinite scroll feature.
+	add_filter( 'jetpack_disable_eu_cookie_law_widget', '__return_true', 1 );
+	add_filter( 'wpcom_disable_logged_out_follow', '__return_true', 10, 1 ); // Disable follow actionbar.
+	add_filter( 'wpl_is_enabled_sitewide', '__return_false', 10, 1 ); // Disable likes.
+	add_filter( 'jetpack_implode_frontend_css', '__return_false', 99 ); // Jetpack "implodes" all registered CSS files into one file.
+	add_filter( 'woocommerce_demo_store', '__return_false' ); // Prevent the the wocommerce demo store notice from displaying.
+	add_filter( 'jetpack_open_graph_image_default', 'A8C\FSE\Coming_soon\coming_soon_share_image' ); // Set the default OG image.
+	add_filter( 'jetpack_twitter_cards_image_default', 'A8C\FSE\Coming_soon\coming_soon_share_image' ); // Set the default Twitter Card image.
```

Comment:
In case we do decide to change the namespace,
```suggestion
	add_filter( 'jetpack_open_graph_image_default', __NAMESPACE__ . '\coming_soon_share_image' ); // Set the default OG image.
	add_filter( 'jetpack_twitter_cards_image_default', __NAMESPACE__ . '\coming_soon_share_image' ); // Set the default Twitter Card image.
```